integrations/Drip/browser.js

```diff
@@ -0,0 +1,161 @@
+import get from "get-value";
+import logger from "../../utils/logUtil";
+import { removeUndefinedAndNullValues } from "../utils/commonUtils";
+
+class Drip {
+  constructor(config) {
+    this.accountId = config.accountId;
+    this.campaignId = config.campaignId;
+    this.name = "DRIP";
+  }
+
+  init() {
+    logger.debug("===In init Drip===");
+
+    window._dcq = _dcq || [];
+    window._dcs = _dcs || {};
+    window._dcs.account = this.accountId;
+
+    (function () {
+      var dc = document.createElement("script");
+      dc.type = "text/javascript";
+      dc.async = true;
+      dc.src = `//tag.getdrip.com/${this.accountId}.js`;
+      var s = document.getElementsByTagName("script")[0];
+      s.parentNode.insertBefore(dc, s);
+    })();
+  }
+
+  identify(rudderElement) {
+    logger.debug("===In Drip identify===");
+
+    const { message } = rudderElement;
+    if (!message.context.traits) {
+      logger.error("user traits not present");
+      return;
+    }
+
+    const { email } = message.context.traits;
+    if (!email) {
+      logger.error("email is required for the call");
+      return;
+    }
+
+    const { euConsent } = message.context.traits;
+    if (
+      euConsent &&
+      !(
+        euConsent.toLowercase() === "granted" ||
+        euConsent.toLowercase() === "denied"
+      )
+    ) {
+      euConsent = "";
+    }
+
+    let payload = {
+      email: email,
+      new_email: get(message, "context.traits.newEmail"),
+      user_id: get(message, "userId"),
+      tags: get(message, "context.traits.tags"),
+      remove_tags: get(message, "context.traits.removeTags"),
+      prospect: get(message, "context.traits.prospect"),
+      eu_consent: euConsent,
+      eu_consent_message: get(message, "context.traits.euConsentMessage"),
+      success: (response) => {
+        // Call a method with the response object
+        // Success callback is optional
+        logger.debug("identify call was success");
+      },
+    };
+    payload = removeUndefinedAndNullValues(payload);
+
+    const campaignId =
+      get(message, "context.traits.campaignId") || this.campaignId;
```

Comment:
lets discuss this once. We probably should not do `get(message, "context.traits.campaignId")` because its not a field as per the drip docs. In cloud mode we are taking it from dashboard or from `context.externalId`. We should do the same here as well